pkg/backup/resource_backupper.go

```diff
@@ -297,34 +296,40 @@ func (rb *defaultResourceBackupper) backupItem(
 	return backedUpItem
 }
 
-// Adds CRD to the backup if one is found corresponding to this resource
-func (rb *defaultResourceBackupper) backupCRD(
-	log logrus.FieldLogger,
-	gr schema.GroupResource,
-	itemBackupper ItemBackupper,
-) {
-	crdGr := schema.GroupResource{Group: apiextv1beta1.GroupName, Resource: "customresourcedefinitions"}
-	crdClient, err := rb.dynamicFactory.ClientForGroupVersionResource(apiextv1beta1.SchemeGroupVersion,
-		metav1.APIResource{
-			Name:       "customresourcedefinitions",
-			Namespaced: false,
-		},
-		"",
-	)
+// backupCRD checks if the resource is a custom resource, and if so, backs up the custom resource definition
+// associated with it.
+func (rb *defaultResourceBackupper) backupCRD(log logrus.FieldLogger, gr schema.GroupResource, itemBackupper ItemBackupper) {
+	log.Debug("Getting server preferred API version for customresourcedefinitions.apiextensions.k8s.io")
+	gvr, apiResource, err := rb.discoveryHelper.ResourceFor(schema.GroupVersionResource{Group: "apiextensions.k8s.io", Resource: "customresourcedefinitions"})
```

Comment:
Now that #1937 is merged into master, you can use `kuberesource.CustomResourceDefinitions` as the argument to `ResourceFor`.

Reply:
ah yeah, i made a mental note of this before and forgot :) thx

Reply:
rebased and added a new commit with the change (no other changes to previous commits).

@carlisia you'll need to review again too, thx.